### PR TITLE
🐛 Fix portfolio & dashboard OG

### PR DESCRIPTION
**Pull request by nwingt**
Seems `head()` inside mixin is troublesome

- [x] Use route `id` for getting the wallet instead of the copy in the component's `data()`
- [x] Reuse user info from the store instead of the copy in the component's `data()`

### CODE REVIEW

**Review by williamchong — CHANGES_REQUESTED** [review]
still not sure why og is broken...

File: src/pages/_id/index.vue [hunk 1]
```diff
@@ -221,44 +270,33 @@ export default {
...
       }
     }
     if (id && checkUserNameValid(id)) {
       try {
         const userInfo = await $api.$get(getUserMinAPI(id));
-        return {
-          userInfo,
-          wallet: userInfo.likeWallet,
-        };
+        redirect({ ...route, params: { id: userInfo.likeWallet } });
```
[the remark this answers is not in the document]
Comment (nwingt) [reply src/pages/_id/index.vue:292]: Added

File: src/pages/_id/index.vue [hunk 2]
```diff
@@ -221,44 +270,34 @@ export default {
...
+          await store.dispatch('fetchUserInfoByAddress', id);
+          return;
         } catch (error) {
           // eslint-disable-next-line no-console
           console.error(error);
         }
-        return {
-          wallet: id,
-          userInfo: undefined,
-        };
```
Comment (williamchong) [inline src/pages/_id/index.vue:244]: Why is this removed? like1 without likerId should be valid user, or another `return` is needed here?

**Review by williamchong — CHANGES_REQUESTED** [review]
Please fix lint error also

**Review by williamchong — APPROVED** [review]
